handle ddl event failed while reorganize partition table

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. create partition table with some data
2. reorganize partition table 
<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)
there are some stats error log
```log
[2023/03/13 12:58:53.376 +00:00] [ERROR] [domain.go:2167] ["handle ddl event failed"] [event="(Event Type: alter table reorganize partition, Table ID: 720, Table Name zto_ssmx_bill_detail, Partition IDs: [721 722 723 724 725 726 727 728 729 730 731 732 733 734 735 736 737 738 739 740 741 742 743 744 745 746 747 748 749 750 751 752 753 754 755 756 757 758 759 760 761 762 763 764 765 766 767 768 769 770 771 772 773 774 775 776 777 778 779 780 781]"] [error="[kv:1062]Duplicate entry '721' for key 'stats_meta.tbl'"]
```
while `penw` is the new reorganized partition
```log
[2023/03/13 12:50:31.912 +00:00] [INFO] [backfilling.go:371] ["[ddl] backfill worker finish task"] [worker="backfill-worker 3, tp add index"] [task="taskID: 7, physicalTableID: 788, range: [7480000000000003145f72a9
[2023/03/13 13:01:46.675 +00:00] [ERROR] [analyze_global_stats.go:79] ["merge global stats failed"] [info="merge global stats for rtdb.zto_ssmx_bill_detail's index idx_ssmx_bill_detail_i003"] [error="[types:8131]Build global-level stats failed due to missing partition-level stats: table `zto_ssmx_bill_detail` partition `pnew`"] [tableID=720]
[2023/03/13 13:01:48.084 +00:00] [INFO] [2pc.go:677] ["[BIG_TXN]"] [session=0] ["key sample"=7480000000000000165f69800000000000000104061b6c0c5c38000403800000000000758a] [size=3531362] [keys=95521] [puts=3] [dels=95517] [locks=1] [checks=0] [txnStartTS=440064411344306179]
[2023/03/13 13:01:48.489 +00:00] [INFO] [gc.go:88] ["remove stats in GC due to dropped table"] [table_id=732]
[2023/03/13 13:01:49.872 +00:00] [INFO] [2pc.go:677] ["[BIG_TXN]"] [session=0] ["key sample"=7480000000000000165f69800000000000000104061b6c207330000303800000000000758b] [size=3535580] [keys=95635] [puts=3] [dels=95631] [locks=1] [checks=0] [txnStartTS=440064411816427530]
[2023/03/13 13:01:50.438 +00:00] [ERROR] [analyze_global_stats.go:79] ["merge global stats failed"] [info="merge global stats for rtdb.zto_ssmx_bill_detail's index idx_ssmx_bill_detail_i004"] [error="[types:8131]Build global-level stats failed due to missing partition-level stats: table `zto_ssmx_bill_detail` partition `pnew`"] [tableID=720]
[2023/03/13 13:01:50.439 +00:00] [INFO] [gc.go:88] ["remove stats in GC due to dropped table"] [table_id=734]
[2023/03/13 13:01:51.566 +00:00] [INFO] [advancer.go:190] ["[log backup advancer hint] current last region"] [min="([6D, 6D44423A3531380000FD0000000000000068), 440064388656005121)"] [for-polling=0] [min-ts=2023-03-13T13:00:20Z]
[2023/03/13 13:01:51.568 +00:00] [INFO] [advancer.go:410] ["updated log backup GC safe point."] [checkpoint=440062836632190976] [target=440064388656005120]
[2023/03/13 13:01:52.257 +00:00] [INFO] [2pc.go:677] ["[BIG_TXN]"] [session=0] ["key sample"=7480000000000000165f69800000000000000104061b6ba36840000403800000000000758d] [size=3541426] [keys=95793] [puts=3] [dels=95789] [locks=1] [checks=0] [txnStartTS=440064412327346184]
[2023/03/13 13:01:52.644 +00:00] [INFO] [gc_worker.go:1199] ["[gc worker] start resolve locks"] [uuid=61b6683340c0013] [safePoint=0] [try-resolve-locks-ts=440064334261125128] [concurrency=1]
[2023/03/13 13:01:52.645 +00:00] [INFO] [range_task.go:137] ["range task started"] [name=resolve-locks-runner] [startKey=] [endKey=] [concurrency=1]
[2023/03/13 13:01:52.645 +00:00] [INFO] [gc.go:88] ["remove stats in GC due to dropped table"] [table_id=735]
[2023/03/13 13:01:53.298 +00:00] [INFO] [range_task.go:246] ["range task finished"] [name=resolve-locks-runner] [startKey=] [endKey=] ["cost time"=653.839132ms] ["completed regions"=801]
[2023/03/13 13:01:53.299 +00:00] [INFO] [gc_worker.go:1221] ["[gc worker] finish resolve locks"] [uuid=61b6683340c0013] [safePoint=0] [try-resolve-locks-ts=440064334261125128] [regions=801]
[2023/03/13 13:01:54.258 +00:00] [INFO] [2pc.go:677] ["[BIG_TXN]"] [session=0] ["key sample"=7480000000000000165f69800000000000000104061b6ba43bf8000503800000000000758e] [size=3533952] [keys=95591] [puts=3] [dels=95587] [locks=1] [checks=0] [txnStartTS=440064412930277378]
[2023/03/13 13:01:54.565 +00:00] [INFO] [gc.go:88] ["remove stats in GC due to dropped table"] [table_id=736]
[2023/03/13 13:01:54.588 +00:00] [ERROR] [analyze_global_stats.go:79] ["merge global stats failed"] [info="merge global stats for rtdb.zto_ssmx_bill_detail's index idx_ssmx_bill_detail_h013"] [error="[types:8131]Build global-level stats failed due to missing partition-level stats: table `zto_ssmx_bill_detail` partition `pnew`"] [tableID=720]
[2023/03/13 13:01:56.088 +00:00] [INFO] [2pc.go:677] ["[BIG_TXN]"] [session=0] ["key sample"=7480000000000000165f69800000000000000104061b6ba5c8d8000503800000000000758f] [size=3540538] [keys=95769] [puts=3] [dels=95765] [locks=1] [checks=0] [txnStartTS=440064413415243787]
[2023/03/13 13:01:56.405 +00:00] [INFO] [gc.go:88] ["remove stats in GC due to dropped table"] [table_id=737]
[2023/03/13 13:01:57.933 +00:00] [INFO] [2pc.go:677] ["[BIG_TXN]"] [session=0] ["key sample"=7480000000000000165f69800000000000000104061b6ba50a380006038000000000007590] [size=3529882] [keys=95481] [puts=3] [dels=95477] [locks=1] [checks=0] [txnStartTS=440064413913579524]
[2023/03/13 13:01:58.244 +00:00] [ERROR] [analyze_global_stats.go:79] ["merge global stats failed"] [info="merge global stats for rtd
```
### 4. What is your TiDB version? (Required)
```sql
| tidb_version()                                            |
+-----------------------------------------------------------+
| Release Version: v6.7.0-alpha                             |
| Edition: Community                                        |
| Git Commit Hash: 145b7cdf72feeecf7f2b0b3e01693c603712dce5 |
| Git Branch: heads/refs/tags/v6.7.0-alpha                  |
| UTC Build Time: 2023-03-11 11:42:07                       |
| GoVersion: go1.20.2                                       |
| Race Enabled: false                                       |
| TiKV Min Version: 6.2.0-alpha                             |
| Check Table Before Drop: false                            |
| Store: tikv                                               |
+-----------------------------------------------------------+
```
<!-- Paste the output of SELECT tidb_version() -->

--- a/ddl/partition.go
+++ b/ddl/partition.go
@@ -2498,7 +2498,7 @@ func (w *worker) onReorganizePartition(d *ddlCtx, t *meta.Meta, job *model.Job) 
 		physicalTableIDs := getPartitionIDsFromDefinitions(tblInfo.Partition.DroppingDefinitions)
 		newIDs := getPartitionIDsFromDefinitions(partInfo.Definitions)
 		job.CtxVars = []interface{}{physicalTableIDs, newIDs}
-		definitionsToDrop := tblInfo.Partition.DroppingDefinitions
+		definitionsToAdd := tblInfo.Partition.AddingDefinitions
 		tblInfo.Partition.DroppingDefinitions = nil
 		tblInfo.Partition.AddingDefinitions = nil
 		ver, err = updateVersionAndTableInfo(d, t, job, tblInfo, true)
@@ -2516,7 +2516,7 @@ func (w *worker) onReorganizePartition(d *ddlCtx, t *meta.Meta, job *model.Job) 
 		// How to handle this?
 		// Seems to only trigger asynchronous update of statistics.
 		// Should it actually be synchronous?
-		asyncNotifyEvent(d, &util.Event{Tp: model.ActionReorganizePartition, TableInfo: tblInfo, PartInfo: &model.PartitionInfo{Definitions: definitionsToDrop}})
+		asyncNotifyEvent(d, &util.Event{Tp: model.ActionReorganizePartition, TableInfo: tblInfo, PartInfo: &model.PartitionInfo{Definitions: definitionsToAdd}})
 		// A background job will be created to delete old partition data.
 		job.Args = []interface{}{physicalTableIDs}
 

--- a/statistics/handle/ddl.go
+++ b/statistics/handle/ddl.go
@@ -68,14 +68,7 @@ func (h *Handle) HandleDDLEvent(t *util.Event) error {
 				return err
 			}
 		}
-		// Update global stats, even though it should not have changed,
-		// the updated statistics from the newly reorganized partitions may be better
-		pruneMode := h.CurrentPruneMode()
-		if pruneMode == variable.Dynamic && t.PartInfo != nil {
-			if err := h.updateGlobalStats(t.TableInfo); err != nil {
-				return err
-			}
-		}
+		// Do not update global stats, since the data have not changed!
 	case model.ActionFlashbackCluster:
 		return h.updateStatsVersion()
 	}

--- a/statistics/handle/ddl_test.go
+++ b/statistics/handle/ddl_test.go
@@ -18,7 +18,6 @@ import (
 	"testing"
 
 	"github.com/pingcap/tidb/parser/model"
-	"github.com/pingcap/tidb/sessionctx/variable"
 	"github.com/pingcap/tidb/testkit"
 	"github.com/pingcap/tidb/types"
 	"github.com/pingcap/tidb/util/mock"
@@ -190,7 +189,9 @@ func TestDDLHistogram(t *testing.T) {
 func TestDDLPartition(t *testing.T) {
 	store, do := testkit.CreateMockStoreAndDomain(t)
 	testKit := testkit.NewTestKit(t, store)
-	testkit.WithPruneMode(testKit, variable.Static, func() {
+	for _, pruneMode := range []string{"static", "dynamic"} {
+		testKit.MustExec("set @@tidb_partition_prune_mode=`" + pruneMode + "`")
+		testKit.MustExec("set global tidb_partition_prune_mode=`" + pruneMode + "`")
 		testKit.MustExec("use test")
 		testKit.MustExec("drop table if exists t")
 		createTable := `CREATE TABLE t (a int, b int, primary key(a), index idx(b))
@@ -261,5 +262,20 @@ PARTITION BY RANGE ( a ) (
 			statsTbl := h.GetPartitionStats(tableInfo, def.ID)
 			require.False(t, statsTbl.Pseudo)
 		}
-	})
+
+		reorganizePartition := "alter table t reorganize partition p0,p1 into (partition p0 values less than (11))"
+		testKit.MustExec(reorganizePartition)
+		is = do.InfoSchema()
+		tbl, err = is.TableByName(model.NewCIStr("test"), model.NewCIStr("t"))
+		require.NoError(t, err)
+		tableInfo = tbl.Meta()
+		err = h.HandleDDLEvent(<-h.DDLEventCh())
+		require.NoError(t, err)
+		require.Nil(t, h.Update(is))
+		pi = tableInfo.GetPartitionInfo()
+		for _, def := range pi.Definitions {
+			statsTbl := h.GetPartitionStats(tableInfo, def.ID)
+			require.False(t, statsTbl.Pseudo)
+		}
+	}
 }

